### ISSUE TITLE
Compatibility with libtorch C++ API

### ISSUE THREAD

**Issue by JSLJ23** [teddykoker/e3nn.c#13]
Hi, really cool work going on here on porting the `e3nn` logic to pure C. Just wondering will there be any future plans to support compatibility with the libtorch C++ API to allow for building equivariant models in pure torch C++?

I was looking to build some equivariant models that need to run in an environment without python so using the original e3nn-pyTorch or e3nn-Jax might not be possible.

If there might not be any plans to support libtorch compatibility, maybe are there any pointers / advice you could give on how something like this could be similarly implemented with libtorch C++ to take full advantage of autograd for building models?

**Comment by mitkotak** [comment]
Thanks for the question !

Does the Torch2->C++ inference pipeline here help (https://github.com/atomicarchitects/e3nn_tetris) or did you also wanna train the model in C++ ?

**Comment by JSLJ23** [comment]
Yes ideally I need the whole training and inference to be in C++ as the model is going to be part of a larger code base that runs QM/NN/MM simulations.

**Comment by teddykoker** [comment]
Hi @JSLJ23 I have no plans regarding compatibility with the libtorch C++ API. I have no experience with it but it looks like it may be possible to [train models that are exported with TorchScript in C++](https://github.com/pytorch/pytorch/issues/17614), so that could be one way to do things. Otherwise if you only need inference the C++ AOTInductor pipeline that @mitkotak mentioned could work (although I don't believe training is supported at the moment). Outside of that you could implement the e3nn operations using the C++ front end, and you might be able to borrow some of the ideas from e3nn.c.
